Changes in Endorser Protocol

HarshMultani-AyanWorks opened the pull request:
This is a WIP pull request, and it covers the changes in endorser protocol for issues defined in 

1. #1031 
2. #1029

ianco commented:
Hi @HarshMultani-AyanWorks , it's hard to tell from the code, but the default behaviour should be that the Author writes the transaction to the ledger and updates the schema/cred def non-secrets record to their own wallet.  There should be an *option* (maybe a new parameter in the `create-request` endpoint) to ask the Endorser to write the ledger transaction (in this scenario the Author still needs to write the non-secrets record to their wallet).  It looks like this update moves *all* of this functionality to the Endorser?  (Unless I'm not reading the code properly?)  Let me know once you've updated the integration test, it's broken now so difficult to review ...

HarshMultani-AyanWorks commented:
> Hi @HarshMultani-AyanWorks , it's hard to tell from the code, but the default behaviour should be that the Author writes the transaction to the ledger and updates the schema/cred def non-secrets record to their own wallet. There should be an _option_ (maybe a new parameter in the `create-request` endpoint) to ask the Endorser to write the ledger transaction (in this scenario the Author still needs to write the non-secrets record to their wallet). It looks like this update moves _all_ of this functionality to the Endorser? (Unless I'm not reading the code properly?) Let me know once you've updated the integration test, it's broken now so difficult to review ...

Hi @ianco,
I have fixed the integration test.
Also updated the unit tests.

I checked and tested the code, the author and endorser, both can invoke the endpoint to write the transaction to the ledger. However, the transaction is written to the ledger from the author's agent and so the records are being stored in author's wallet.
Can you point me, if I am missing something?

sklump commented on aries_cloudagent/protocols/endorse_transaction/v1_0/handlers/tests/test_transaction_acknowledgement_handler.py:
```diff
@@ -0,0 +1,84 @@
+
+from ...handlers import transaction_acknowledgement_handler as test_module
+from ...messages.transaction_acknowledgement import TransactionAcknowledgement
+from ......connections.models.conn_record import ConnRecord
```
There is an order to follow, for future reference:
* import native package[s]
* one blank line
* from native package[s] import item[s]
* one blank line
* import third party package[s]
* one blank line
* from third party package[s] import item[s]
* one blank line
* from relative package[s] import item[s]: farthest to closest, one blank line between levels
* alphabetical order within each grouping above.

For now it's fine and I will get to this on some future pass

* from farthest to closest

sklump commented on aries_cloudagent/protocols/endorse_transaction/v1_0/tests/test_manager.py:
```diff
@@ -1,4 +1,5 @@
 import uuid
+import json
```
flip them

sklump commented on aries_cloudagent/protocols/endorse_transaction/v1_0/tests/test_manager.py:
```diff
@@ -16,6 +17,15 @@
 from ..messages.transaction_request import TransactionRequest
+from ..messages.transaction_acknowledgement import TransactionAcknowledgement
+from ..transaction_jobs import TransactionJob
+from .....ledger.base import BaseLedger
```
... see? You've duplicated the TransactionJob import.
